kview/src/main/java/dev/ikm/komet/kview/mvvm/model/DataModelHelper.java

```diff
@@ -77,4 +104,90 @@ public static Set<ConceptEntity> fetchDescendentsOfConcept(ViewProperties viewPr
                 .collect(Collectors.toSet());
         return allDecendents;
     }
+
+    public static List<PatternEntityVersion> getMembershipPatterns() {
+        List<PatternEntityVersion> membershipPatternList = new ArrayList<>();
+        PrimitiveData.get().forEachPatternNid((patternNid) -> {
+            PatternEntityVersion patternEntityVersion = (PatternEntityVersion)
+                    Calculators.View.Default().stampCalculator().latest(patternNid).get();
+            if (patternEntityVersion.semanticPurposeNid() == TinkarTerm.MEMBERSHIP_SEMANTIC.nid()) {
+                membershipPatternList.add(patternEntityVersion);
+            }
+        });
+        return membershipPatternList;
+    }
+
+    public static boolean isInMembershipPattern(int conceptNid, int patternNid) {
```

Comment:
Can the two arguments be PublicIds instead of nids?
The goal is to use the protocol buffer objects (they don't contain nids, but they do contain PublicIds).

Reply:
not without a different API call... that method calls for for the nids... this would have to be a discussion the data team
![image](https://github.com/user-attachments/assets/2da3714a-cfce-49f9-9015-2f3ad9b977e6)